src/coreclr/vm/i386/excepx86.cpp

```diff
@@ -3069,29 +3075,36 @@ void ResumeAtJitEH(CrawlFrame* pCf,
         _ASSERTE(pExInfo->m_pPrevNestedInfo == 0 || pExInfo->m_pPrevNestedInfo->m_StackAddress >= dEsp);
 
         // Before we unwind the SEH records, get the Frame from the top-most nested exception record.
-        Frame* pNestedFrame = GetCurrFrame(FindNestedEstablisherFrame(GetCurrentSEHRecord()));
+        PEXCEPTION_REGISTRATION_RECORD nestedRecord = FindNestedEstablisherFrame(GetCurrentSEHRecord());
+        Frame* pNestedFrame = GetCurrFrame(nestedRecord);
```

Comment:
At like 3112 in the updated code, this `pNestedFrame` is passed into `pNestedHandlerExRecord->Init(...` With the update to the assert I think `pNestedFrame` will be pointing into random stack memory. What is that used for? Could we just set it to `NULL` or something?

Reply:
The `pFrame` isn't random. It is immediately before the added `pNestedHandlerExRecord` &ndash; at least in the scenario in question. That is pointing at the beginning of managed function frame that is containing the try/catch.